### DISCUSSION TITLE
Wyze AI events/alarms

### DISCUSSION THREAD

**Discussion by devinslick** [gtxaspec/wz_mini_hacks#323] [Q&A]
Hey all,

I've been tinkering with this for a few hours now and wanted to reach out to see if others have attempted/succeeding in getting something like this working.

Goals:
1) Be able to poll the latest Wyze detection images via HTTP.  
2) Identify Person/Pet/Package detection events within those alarm events/images
3) Configure the WyzeCam v3 to push a notification to Home Assistant when an alarm/detection occurs.  This could just be a shell script that copies info via SCP or calls a webhook

Progress:
1) Located path to alarm/detection images.  /media/mmc/alarm/YYYYMMDD/YYYYMMDD_HH_MM_SS.jpg.  I've been unable to find a way to list and download these using the httpd server.  
2) 'EdgeAI' binaries and configuration are located in /tmp/ and define the parameters and models to use in the process execution.  "/tmp/edgeai/wyzeedgeai_cam_v3_prod /tmp/edgeai/aiparams.ini /tmp/edgeai/model_params.ini". Now, if there's a way to see the results of this process?
3) I've considered trying to get inotifyd setup to watch the alarm directory, but this won't help with differentiating types of alarms. There is probably a better way to inject a script into the process that's already aware of the events rather than monitoring the folder after events are created.

I'm going to continue working on this, but if anyone else has solutions or ideas here I'd love to hear them.

Thanks!

**Comment by virmaior** [comment]
for 1, the httpd server only looks at `/opt/wz_mini/www/` and beneath, you can write an (b)ash file and put it in `/opt/wz_mini/www/cgi-bin/` that could expose the jpgs you are looking for. This is pretty easy to do (I haven't used this sort of feature myself on my V3s), so I can't really test it.


for 2 and 3, I've got no ida.

**Reply by devinslick** [reply]
Well, I've been poking at this for a while and I've haven't been able to get categorized AI events, but I have gotten motion events pushed to Home Assistant via webhooks - and the latest motion image/video usable as a Generic Camera over HTTP:

http://camera.domain.local/latest.jpg
http://camera.domain.local/latest.mp4

Here's the script with install instructions. 
https://github.com/devinslick/wyze-motion-latest/blob/main/latest.sh

Any ideas for improving this?

**Reply by mrlt8** [reply]
I believe the camera dumps (or used to dump) the files to  /tmp before copying it to the SD card. You could try monitoring `/tmp/` since it may get the files a little earlier:
https://github.com/FiveLeavesLeft/WyzeCameraLiveStream/issues/5#issuecomment-854913734

**Reply by devinslick** [reply]
Since I'm not sure how to watch that directory and grab files after their handles have been closed, but before they've been deleted, I'm not sure how the temp directory can be used reliably.

I've updated the script to loop, and it catches alarms and videos almost immediately and notifies Home Assistant.

There is probably a more efficient way of getting the events but this has been working reliably for me on 12 cameras.

**Comment by jrbecart** [comment]
> 2\. Identify Person/Pet/Package detection events within those alarm events/images

I am not an expert but if the post-process can be done outside the camera, maybe you can try some of these:
https://github.com/tryolabs/luminoth
https://github.com/jacobgil/pytorch-grad-cam
https://github.com/blakeblackshear/frigate (not sure if it can be used on images only)
https://github.com/pytorch/pytorch (https://pyimagesearch.com/2021/08/02/pytorch-object-detection-with-pre-trained-networks/)
https://github.com/ultralytics/yolov5
https://github.com/matterport/Mask_RCNN
https://github.com/open-mmlab/mmdetection
https://github.com/AlexeyAB/darknet
